Практическая работа к 9му уроку.

Kederly84 opened the pull request:
Добрый день.
Выполнены все задания.
Сомневаюсь, что верно понял условия задач, т.к. они кажутся какими то через чур простыми.

Kederly84 committed:
Добрый день.
Выполнены все задания.
Сомневаюсь, что верно понял условия задач, т.к. они кажутся какими то через чур простыми.

shishkinav commented on Lesson_9/Aleksandr_Maksimov_Task_9_2.py:
```diff
@@ -0,0 +1,19 @@
+    _width: int = 0
+    # Определение необходимых атрибутов при создании экземпляра класса
+    def __init__(self, lenght: int, widht: int):
+        Road._lenght = lenght
+        Road._width = widht
```
решили хранить в глобальных переменных?
здесь логично было бы их определенить непосредственно только для создаваемого экземпляра, т.е. в `__init__` как `self._lenght`, `self._width`

shishkinav commented on Lesson_9/Aleksandr_Maksimov_Task_9_2.py:
```diff
@@ -0,0 +1,19 @@
+    def mass(self, coating_thickness: int):
+        """Метод для расчета массы асфальта"""
+        mass_of_layer = 25
+        total_mass = int((Road._width * Road._lenght * mass_of_layer * coating_thickness) / 1000)
```
по заданию вроде не сказано отбросить издержки, округлив до целого ))

shishkinav commented on Lesson_9/Aleksandr_Maksimov_Task_9_3.py:
```diff
@@ -0,0 +1,31 @@
+# Создание класса
+class Worker:
+    name: str = 'Александр'
+    surname: str = 'Максимов'
+    position: str = 'Дворник'
+    _income: dict = {'wage': 0,
```
интересную Вы себе должность придумали, но вот это как раз бы в конструктор передавать и внутри него в переменные экземпляра сохранять; Вы немного недопоняли назначение глобальных переменных в классах - постараюсь незабыть на вебинаре проговорить

shishkinav commented on Lesson_9/Aleksandr_Maksimov_Task_9_3.py:
```diff
@@ -0,0 +1,31 @@
+    _income: dict = {'wage': 0,
+                     'bonus': 0}
+
+    def __init__(self, wage: int, bonus: int):
```
`wage: int, bonus: int` - а вот это как раз не надо было бы дополнительно передавать переменными, они попали бы в экземпляр в виде готового словаря для `income`

shishkinav commented on Lesson_9/Aleksandr_Maksimov_Task_9_3.py:
```diff
@@ -0,0 +1,31 @@
+
+    def get_full_name(self):
+        """Метод для вывода информации по работнику"""
+        print(f'{Position.name} {Position.surname}')
```
неплохо бы добавить `.title()` на случай если окажется, что имя или фамилия с нарушениями регистра внесены

shishkinav commented on Lesson_9/Aleksandr_Maksimov_Task_9_3.py:
```diff
@@ -0,0 +1,31 @@
+
+    def get_total_income(self, seniority: int):
+        """Метод для расчета дохода сотрудника за заданный период"""
+        inc = seniority * (Position._income['wage'] + Position._income['bonus'])
```
`sum(Position._income.values())` ??

shishkinav commented on Lesson_9/Aleksandr_Maksimov_Task_9_4.py:
```diff
@@ -0,0 +1,105 @@
+
+# Создание родительского класса
+class Car:
+    speed: int = 0
+    color: str = 'red'
+    name: str = 'Запорожец'
+    is_police: bool = False
```
выносить в конструктор, иначе получается, что у Вас одна общая для всех экземпляров скорость, цвет, название - говорю о, например, двух объектах созданных из одного класса `TownCar` - можете попробовать ))

shishkinav commented on Lesson_9/Aleksandr_Maksimov_Task_9_5.py:
```diff
@@ -0,0 +1,39 @@
+# Создание родительского класса
+class Stationery:
+    title: str = 'Канцелярская принадлежность'
+    def draw(self):
+        """Метод отрисовки"""
+        print(f'Запуск отрисовки')
```
а ещё можно "заставить" при наследовании от этого класса всех потомков обязательно переопределять метод - для этого используют абстрактные классы
```
from abc import ABC, abstractmethod

class Stationery(ABC):

    def __init__(self, title: str):
        self.title = title

    @abstractmethod
    def draw(self):
        return print('"Запуск отрисовки"')
```

Kederly84 committed:
Добрый день.
Выполнены все задания.
Сомневаюсь, что верно понял условия задач, т.к. они кажутся какими то через чур простыми.

Kederly84 committed:
Добрый день.
Выполнены все задания.
Сомневаюсь, что верно понял условия задач, т.к. они кажутся какими то через чур простыми.

Kederly84 committed:
Добрый день.
Выполнены все задания.
Сомневаюсь, что верно понял условия задач, т.к. они кажутся какими то через чур простыми.
